enterprise/app/sidebar/sidebar.tsx

```diff
@@ -154,80 +155,87 @@ export default class SidebarComponent extends React.Component<Props, State> {
-            <span className="sidebar-item-text">All builds</span>
-          </SidebarLink>
-          <SidebarLink selected={this.isTrendsSelected()} href={Path.trendsPath} title="Trends">
-            <BarChart2 className="icon" />
-            <span className="sidebar-item-text">Trends</span>
-          </SidebarLink>
-          {capabilities.test && (
-            <SidebarLink selected={this.isTapSelected()} href={Path.tapPath} title="Tests">
-              <LayoutGrid className="icon" />
-              <span className="sidebar-item-text">Tests</span>
-            </SidebarLink>
-          )}
-          <SidebarLink selected={this.isUsersSelected()} href="/#users" title="Users">
-            <Users className="icon" />
-            <span className="sidebar-item-text">Users</span>
-          </SidebarLink>
-          <SidebarLink selected={this.isReposSelected()} href="/#repos" title="Repos">
-            <Github className="icon" />
-            <span className="sidebar-item-text">Repos</span>
-          </SidebarLink>
-          <SidebarLink selected={this.isBranchesSelected()} href="/#branches" title="Branches">
-            <GitBranch className="icon" />
-            <span className="sidebar-item-text">Branches</span>
-          </SidebarLink>
-          <SidebarLink selected={this.isCommitsSelected()} href="/#commits" title="Commits">
-            <GitCommit className="icon" />
-            <span className="sidebar-item-text">Commits</span>
-          </SidebarLink>
-          <SidebarLink selected={this.isHostsSelected()} href="/#hosts" title="Hosts">
-            <HardDrive className="icon" />
-            <span className="sidebar-item-text">Hosts</span>
-          </SidebarLink>
-          {router.canAccessExecutorsPage(this.props.user) && (
-            <SidebarLink selected={this.isExecutorsSelected()} href={Path.executorsPath} title="Executors">
-              <Cloud className="icon" />
-              <span className="sidebar-item-text">Executors</span>
-            </SidebarLink>
-          )}
-          {router.canAccessWorkflowsPage(this.props.user) && (
-            <SidebarLink selected={this.isWorkflowsSelected()} href={Path.workflowsPath} title="Workflows">
-              <PlayCircle className="icon" />
-              <span className="sidebar-item-text">Workflows</span>
-            </SidebarLink>
-          )}
-          {capabilities.code && (
-            <SidebarLink selected={this.isCodeSelected()} href={Path.codePath} title="Code">
-              <Code className="icon" />
-              <span className="sidebar-item-text">Code</span>
-            </SidebarLink>
-          )}
-          <SidebarLink selected={this.isSetupSelected()} href={Path.setupPath} title="Quickstart">
-            <Terminal className="icon" />
-            <span className="sidebar-item-text">Quickstart</span>
-          </SidebarLink>
+          {this.props.user?.selectedGroupAccess === user.SelectedGroup.Access.ALLOWED && (
```

Comment:
Maybe Siggi has thoughts here as well, but to avoid introducing another visibility dimension for each sidebar item, we could also just hide all of the sidebar items if the user doesn't have access.

Reply:
Could we just show the full sidebar? If they click on anything other than the group picker, presumably they'll get the same "Access not allowed" page. Not sure how nice a mostly empty sidebar would look anyway

Reply:
The access check was only being performed once on login. I updated the check to happen on every navigation and reverted the change to hide sidebar items.